Kconfig.zephyr

```diff
@@ -704,6 +704,23 @@ config BUILD_OUTPUT_ADJUST_LMA
 	    default "$(dt_chosen_reg_addr_hex,$(DT_CHOSEN_IMAGE_M4))-\
 	             $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH))"
 
+config BUILD_OUTPUT_ADJUST_LMA_SECTIONS
+	def_string "*"
+	depends on BUILD_OUTPUT_ADJUST_LMA!=""
+	help
+	  This determines the output sections to which the above LMA adjustment
+	  will be applied.
+	  The value can be the name of a section in the final ELF, like "text".
+	  It can also be a pattern with wildcards, such as "*bss", which could
+	  match more than one section name. Multiple such patterns can be given
+	  as a ";"-separated list. It's possible to supply a 'negative' pattern
+	  starting with "!", to exclude sections matched by a preceding pattern.
+
+	  By default, all sections will have their LMA adjusted. The following
+	  example excludes one section produced by the code relocation feature:
+	  config BUILD_OUTPUT_ADJUST_LMA_SECTIONS
+	    default "*;!.extflash_text_reloc"
```

Comment:
The value format is indeed GNU-centric, but I don't necessarily see this as a blocker:
* For LLVM, I don't believe llvm-objcopy supports `--change-section-lma` today, but if it were added, then I'm guessing it would be compatible.
* For the ones that already use CMake wrapper scripts, they can always translate the values freely. They can also throw an error if they recognize some part of a value to be unsupported, e.g., wildcards.

Reply:
> The value format is indeed GNU-centric, but I don't necessarily see this as a blocker:

I didn't say it was a blocker, but we should always be careful in such cases, and always consider if code can be made in a way which abstracts such setting in more generic way.

And keeping values separated often ease later processing.

Reply:
accepted for now, although this probably has to be reworked if toolchains like ARM Compiler 6 or arc metaware is going to work with this.